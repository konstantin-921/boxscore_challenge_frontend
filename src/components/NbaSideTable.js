import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import Grid from '@material-ui/core/Grid'
import Table from '@material-ui/core/Table'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import TableCell from '@material-ui/core/TableCell'
import TableBody from '@material-ui/core/TableBody'

const styles = theme => ({
  tableCell: {
    paddingRight: 5,
    paddingLeft: 5,
    textAlign: 'center',
    border: '0.5px solid #e0e0e0',
    borderLeft: 'none',
    backgroundColor: '#f0f0f0',
    '&:last-child': {
      paddingRight: 5
    },
  },
});

class NbaSideTable extends React.Component {
  render() {
    const { classes } = this.props
    const data = this.props.data
    const pointsAway = data.away_totals.points
    const pointsHome = data.home_totals.points
    return (
      <Grid item xs={1}> 
        <Table>
          <TableHead>
            <TableRow>
              <TableCell className={classes.tableCell}>P</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
              <TableRow>
                <TableCell className={classes.tableCell}>{pointsAway}</TableCell>
              </TableRow>
              <TableRow>
                <TableCell className={classes.tableCell}>{pointsHome}</TableCell>
              </TableRow>
          </TableBody>
        </Table>
      </Grid>
    );
  }
}

NbaSideTable.propTypes = {
  data: PropTypes.shape({
    away_totals: PropTypes.shape({
      points: PropTypes.number.isRequired,
    }).isRequired,
    home_totals: PropTypes.shape({
      points: PropTypes.number.isRequired,
    }).isRequired,
  }).isRequired,
};

export default withStyles(styles)(NbaSideTable)