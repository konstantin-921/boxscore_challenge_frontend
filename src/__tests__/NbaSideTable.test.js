import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import NbaSideTable from '../components/NbaSideTable'

describe('NbaSideTable component', () => {

  const props = {
    away_totals: { points: 90 },
    home_totals: { points: 109 },
  }

  it('renders correctly', () => {
    const tree = renderer
      .create(<NbaSideTable data={props} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  });

})