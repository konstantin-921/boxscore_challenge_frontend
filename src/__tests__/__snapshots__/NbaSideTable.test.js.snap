// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NbaSideTable component renders correctly 1`] = `
<div
  className="MuiGrid-item-3 MuiGrid-grid-xs-1-31"
>
  <table
    className="MuiTable-root-99"
  >
    <thead
      className="MuiTableHead-root-100"
    >
      <tr
        className="MuiTableRow-root-101 MuiTableRow-head-104"
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-head-107 NbaSideTable-tableCell-1"
          scope="col"
        >
          P
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root-118"
    >
      <tr
        className="MuiTableRow-root-101"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 NbaSideTable-tableCell-1"
        >
          90
        </td>
      </tr>
      <tr
        className="MuiTableRow-root-101"
      >
        <td
          aria-sort={null}
          className="MuiTableCell-root-106 MuiTableCell-body-108 NbaSideTable-tableCell-1"
        >
          109
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
